staging/src/k8s.io/client-go/transport/transport.go

```diff
@@ -86,22 +86,22 @@ func TLSConfigFor(c *Config) (*tls.Config, error) {
 	return tlsConfig, nil
 }
 
-// loadTLSFiles copies the data from the CertFile, KeyFile, and CAFile fields into the CertData,
+// loadTLSData copies the data from the CertFile, KeyFile, and CAFile fields into the CertData,
 // KeyData, and CAFile fields, or returns an error. If no error is returned, all three fields are
 // either populated or were empty to start.
-func loadTLSFiles(c *Config) error {
+func loadTLSData(c *Config) error {
 	var err error
-	c.TLS.CAData, err = dataFromSliceOrFile(c.TLS.CAData, c.TLS.CAFile)
+	c.TLS.CAData, err = dataFromSliceFuncOrFile(nil, c.TLS.CAData, c.TLS.CAFile)
 	if err != nil {
 		return err
 	}
 
-	c.TLS.CertData, err = dataFromSliceOrFile(c.TLS.CertData, c.TLS.CertFile)
+	c.TLS.CertData, err = dataFromSliceFuncOrFile(c.TLS.GetCert, c.TLS.CertData, c.TLS.CertFile)
```

Comment:
I expected using [`crypto/tls.Config.GetClientCertificate`](https://golang.org/pkg/crypto/tls/#Config) instead. That way you could actually rotate these on a live transport.

Reply:
I think we also need to close old connections when a rotation occurs. See:

https://github.com/kubernetes/kubernetes/blob/master/pkg/kubelet/certificate/transport.go#L149

Reply:
Implemented GetClientCertificate.
Will work on getting old connections closed on cert rotation next.